### PR TITLE
Setup Scripts Fixed

### PR DESCRIPTION
**Pull request by kamakay**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
* Setup script fixed
### Description of the change
* Setup scripts fixed

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@kamakay__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by kamakay** [comment]
@bmoore-msft : I received a template error but we modified only artifact scripts

File: spark-2.0-on-suse/azuredeploy.json [hunk 1]
```diff
@@ -23,6 +23,57 @@
 	  "minValue": 1,
       "maxValue": 50
     },
+    "sparkSlavesSize": {
```
Comment (bmoore-msft) [inline spark-2.0-on-suse/azuredeploy.json]: let's leave this as a variable - it doesn't take anything away from the sample and makes template easier to maintain...

Comment (kamakay) [inline spark-2.0-on-suse/azuredeploy.json]: Hi, I have seen many different templates implements a "dropdown" when executing template, is it mandatory to use a variable?

Comment (kamakay) [inline spark-2.0-on-suse/azuredeploy.json]: @bmoore-msft: I agree with you, there is also another use case fro training for example, where people would spin-up the template without investigating what could be a specific machine features. Could be t-shirt size another good option?

Comment (bmoore-msft) [inline spark-2.0-on-suse/azuredeploy.json]: If there is a pivot in machine size for this particular sample that's interesting we can always add to the readme...

File: spark-2.0-on-suse/azuredeploy.json [hunk 2]
```diff
@@ -44,8 +44,8 @@
     "storageAccountName": "[concat(uniquestring(resourceGroup().id),'storageac')]",
     "sparkMasterMachineName": "sparkmaster",
     "sparkSlaveMachineName": "sparkslave",
-    "sparkMasterSize": "Basic_A3",
-    "sparkSlavesSize": "Basic_A2",
+    "sparkMasterSize": "Standard_D3_v2",
+	"sparkSlavesSize": "Standard_D3_v2",
```
Comment (bmoore-msft) [inline spark-2.0-on-suse/azuredeploy.json]: There are a few places in the template where we've mixed tabs and spaces for indenting...  Can you throw this in VS Code and format it (SIFT+ALT+F) and then I think we good to go...

File: spark-2.0-on-suse/azuredeploy.json (hunk 1 above)
Comment (kamakay) [inline spark-2.0-on-suse/azuredeploy.json]: Ok, nice idea

File: spark-2.0-on-suse/azuredeploy.json (hunk 2 above)
Comment (kamakay) [inline spark-2.0-on-suse/azuredeploy.json]: Ok, done ☺

Comment (kamakay) [inline spark-2.0-on-suse/azuredeploy.json]: Hi @bmoore-msft this is fixed for my point of view, can you check please?

File: spark-2.0-on-suse/README.md [hunk 3]
```diff
@@ -71,21 +70,64 @@ VMs will be used as spark master or N-th worker on the cluster
...
+You would use for instance a "work in progress" github location; you can specify it when providing template parameters
+<pre>
+artifactsLocation -> https://raw.githubusercontent.com/valueamplify/azure-quickstart-templates/fixsetupscripts/spark-2.0-on-suse/
+</pre>
+
+#### Azure Blob Storage
+
+This spark standalone cluster is already bound to the Azure Blob Storage Account created with the cluster.
+
+You can use a third part tool such as [Microsoft Azure Storage Explorer](http://storageexplorer.com) to connect and create a specific container for your data.
```
Comment (bmoore-msft) [inline spark-2.0-on-suse/README.md]: Storage explorer is 1st party...

File: spark-2.0-on-suse/azuredeploy.json [hunk 4]
```diff
@@ -178,30 +178,30 @@
...
-       ]
+        {
+          "type": "extensions",
+          "name": "configuresparkonslave",
+          "apiVersion": "2015-06-15",
+          "location": "[variables('resourceGroupLocation')]",
+          "dependsOn": [
+            "[concat('Microsoft.Network/networkInterfaces/', concat(variables('sparkSlaveNetworkInterfaceName'), copyindex() ) )]",
+            "[resourceId('Microsoft.Compute/virtualMachines', concat(variables('sparkSlaveMachineName'), copyindex() ) )]",
+            "[concat('Microsoft.Compute/virtualMachines/sparkmaster/extensions/configuresparkonmaster')]"
```
Comment (bmoore-msft) [inline spark-2.0-on-suse/azuredeploy.json]: there's only one arg - nothing to concat to

File: spark-2.0-on-suse/azuredeploy.json [hunk 5]
```diff
@@ -178,30 +178,30 @@
...
-	  }
-	}
-       ]
+        {
+          "type": "extensions",
+          "name": "configuresparkonslave",
+          "apiVersion": "2015-06-15",
+          "location": "[variables('resourceGroupLocation')]",
+          "dependsOn": [
+            "[concat('Microsoft.Network/networkInterfaces/', concat(variables('sparkSlaveNetworkInterfaceName'), copyindex() ) )]",
```
Comment (bmoore-msft) [inline spark-2.0-on-suse/azuredeploy.json]: use resourceId() for consistency

File: spark-2.0-on-suse/README.md (hunk 3 above)
Comment (kamakay) [inline spark-2.0-on-suse/README.md]: ok, sure :)

File: spark-2.0-on-suse/azuredeploy.json (hunk 4 above)
Comment (kamakay) [inline spark-2.0-on-suse/azuredeploy.json]: Ok done

**Comment by kamakay** [comment]
@bmoore-msft can you check please? Thanks

File: spark-2.0-on-suse/azuredeploy.json [hunk 6]
```diff
@@ -308,106 +308,106 @@
...
+        ]
       }
     },
     {
       "type": "Microsoft.Storage/storageAccounts",
       "sku": {
-      	"name": "Standard_LRS"
-      	, "tier": "Standard" },
+        "name": "Standard_LRS"
+        , "tier": "Standard" },
```
Comment (bmoore-msft) [inline spark-2.0-on-suse/azuredeploy.json]: put the comma on the previous line

**Pull request merged**